internal/controller/etcdopstask/handler/ondemandsnapshot/ondemandsnapshot.go

```diff
@@ -0,0 +1,224 @@
+	cl := ptr.Deref(httpClient, http.Client{Timeout: time.Second * time.Duration(snapshotTimeout)})
+
+	return &Handler{
+		client:        k8sClient,
+		etcdReference: etcdRef,
+		httpClient:    cl,
+		config:        *task.Spec.Config.OnDemandSnapshot,
+	}, nil
+}
+
+// Admit checks if the task can be admitted for execution.
+func (h *Handler) Admit(ctx context.Context) handler.Result {
+	var etcd druidv1alpha1.Etcd
+	if err := h.client.Get(ctx, h.etcdReference, &etcd); err != nil {
+		// Check the type of error and return result. If it's a transport layer issue, requeue. If the object is not found, return reject.
+		if apierrors.IsNotFound(err) {
+			return handler.Result{
+				Description: "Failed to get etcd object",
+				Error:       druiderr.WrapError(err, ErrGetEtcd, string(druidv1alpha1.OperationPhaseAdmit), "failed to get etcd object"),
+				Requeue:     false,
+			}
+		}
+		return handler.Result{
+			Description: "Failed to get etcd object due to internal error",
+			Error:       druiderr.WrapError(err, ErrGetEtcd, string(druidv1alpha1.OperationPhaseAdmit), "failed to get etcd object due to internal error"),
+			Requeue:     true,
+		}
+	}
+
+	if !etcd.IsBackupStoreEnabled() {
+		return handler.Result{
+			Description: "Backup is not enabled for etcd",
+			Error:       druiderr.WrapError(fmt.Errorf("backup is not enabled for etcd"), ErrBackupNotEnabled, string(druidv1alpha1.OperationPhaseAdmit), "backup is not enabled for etcd"),
+			Requeue:     false,
+		}
+	}
+
+	if err := etcd.IsReady(); err != nil {
+		return handler.Result{
+			Description: "Etcd is not ready",
+			Error:       druiderr.WrapError(err, ErrEtcdNotReady, string(druidv1alpha1.OperationPhaseAdmit), "etcd is not ready"),
+			Requeue:     false,
+		}
+	}
+	return handler.Result{
+		Description: "Admit check passed",
+		Requeue:     false,
+	}
+}
+
+// Run executes the on-demand snapshot task.
+func (h *Handler) Run(ctx context.Context) handler.Result {
+	etcd := &druidv1alpha1.Etcd{}
+	if err := h.client.Get(ctx, h.etcdReference, etcd); err != nil {
+		if apierrors.IsNotFound(err) {
+			return handler.Result{
+				Description: "Failed to get etcd object - object not found",
+				Error:       druiderr.WrapError(err, ErrGetEtcd, string(druidv1alpha1.OperationPhaseRunning), "failed to get etcd object - object not found"),
+				Requeue:     false,
+			}
+		}
+		return handler.Result{
+			Description: "Failed to get etcd object due to transient error",
+			Error:       druiderr.WrapError(err, ErrGetEtcd, string(druidv1alpha1.OperationPhaseRunning), "failed to get etcd object due to transient error"),
+			Requeue:     true,
+		}
+	}
+	if err := etcd.IsReady(); err != nil {
+		return handler.Result{
+			Description: "Etcd is not ready",
+			Error:       druiderr.WrapError(err, ErrEtcdNotReady, string(druidv1alpha1.OperationPhaseRunning), "etcd is not ready"),
+			Requeue:     false,
+		}
+	}
+	var httpScheme string
+	if tlsConfig := etcd.Spec.Backup.TLS; tlsConfig != nil {
```

Comment:
Please create a separate function that will create a HTTPClient after looking at the relevant secret etc.